### ISSUE TITLE
Card Scanning like Opal Travel and My Opal

### ISSUE THREAD

**Issue by jayenashar** [timgreen/opaler#5]
apparently there's a lot of info you can scan off the card with an NFC-enabled device

**Comment by NeilForsyth** [comment]
Most of the data on an Opal card is encrypted, except for 16 bytes which includes the card number, and a cache of the current balance + details of the most recent transaction.

It's a MiFare DESFire EV1 (`MF3ICD41`) NFC chip, with Application ID `0x534531` used for Opal, 6 AES encrypted files, and file 7 is the one that can be interpreted.

**Comment by jayenashar** [comment]
opal travel also shows if auto top-up is enabled and how many journeys you did this week (maybe also from the card?)  apparently the auto top-up shows as disabled if you have one of the first opal cards.

**Comment by NeilForsyth** [comment]
Yes, that's true. I found a good write-up of the available data at: https://github.com/codebutler/farebot/issues/105

Here's the 16 bytes from my spare card:
'4A C2 77 00 E0 01 A0 63 04 9C CA A0 A4 1B AC F6'

**Comment by micolous** [comment]
Documentation on that bug isn't quite correct from my investigations. I've got code for it in a fork of farebot, with a code implementation (which shows you bit ranges that you need to read).

I haven't implemented all of the fields yet, as I was interested in developing a proof of concept.
